File: packages/emitter/testcases/common/expected/attributes/comprehensive/Attributes.cs
namespace TestCases.common.attributes.comprehensive
{
    [global::System.SerializableAttribute]
    public class User
    {
        [global::System.ObsoleteAttribute("ctor")]
        public User()
        {

        }

        private string _nameField { get; set; }

        [global::System.ObsoleteAttribute("prop")]
        public string name
            {
            get
                {
                    return this._nameField;
                }
            set
                {
                    this._nameField = value;
                }
            }

        [global::System.ObsoleteAttribute("method")]
        public void save()
        {

        }
    }

    public class NoCtor
    {
        [global::System.ObsoleteAttribute("implicit")]
        public NoCtor()
        {

        }

        public double value { get; set; }
    }
}

File: packages/emitter/testcases/common/attributes/comprehensive/Attributes.ts
import { attributes as A } from "@tsonic/core/lang.js";
import {
  ObsoleteAttribute,
  SerializableAttribute,
} from "@tsonic/dotnet/System.js";

export class User {
  #nameField!: string;

  get name(): string {
    return this.#nameField;
  }

  set name(value: string) {
    this.#nameField = value;
  }

  save(): void {}
}

A<User>().add(SerializableAttribute);
A<User>().ctor.add(ObsoleteAttribute, "ctor");
A<User>()
  .method((u) => u.save)
  .add(ObsoleteAttribute, "method");
A<User>()
  .prop((u) => u.name)
  .add(ObsoleteAttribute, "prop");

export class NoCtor {
  value!: number;
}

A<NoCtor>().ctor.add(ObsoleteAttribute, "implicit");
